packages/ckeditor5-image/src/image/imageblocktoinlinecommand.js

```diff
@@ -0,0 +1,68 @@
+/**
+ * @license Copyright (c) 2003-2020, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md or https://ckeditor.com/legal/ckeditor-oss-license
+ */
+
+/**
+ * @module image/image/imageinlinetoblockcommand
+ */
+
+import Command from '@ckeditor/ckeditor5-core/src/command';
+import { findOptimalInsertionPosition } from '@ckeditor/ckeditor5-widget/src/utils';
+import { isImage } from './utils';
+
+/**
+ * The image inline command. It is used to convert block image io inline image.
+ *
+ * @extends module:core/command~Command
+ */
+export default class ImageBlockToInlineCommand extends Command {
+	/**
+	 * @inheritDoc
+	 */
+	refresh() {
+		const element = this.editor.model.document.selection.getSelectedElement();
+
+		this.isEnabled = isImage( element );
+	}
+
+	/**
+	 * Executes the command.
+	 *
+	 * @fires execute
+	 */
+	execute() {
+		const model = this.editor.model;
+		const selection = model.document.selection;
+		const imageElement = selection.getSelectedElement();
+		const src = imageElement.getAttribute( 'src' );
+		const alt = imageElement.getAttribute( 'alt' );
+		const srcset = imageElement.getAttribute( 'srcset' );
+		const position = findOptimalInsertionPosition( selection, model );
```

Comment:
Maybe we don't need to search for an optimal insertion position - the selection is already on the block image widget so we could just replace it with a paragraph with an inline image inside it. Also `model.insertContent` would handle removing old content for us.